src/dom/__tests__/asyncHook.test.ts

```diff
@@ -100,6 +100,54 @@ describe('async hook tests', () => {
 
     expect(complete).toBe(true)
   })
+  test('should wait for arbitrary expectation to pass when use jest.usedFakeTimers() and advanceTimersByTime() ', async () => {
+    jest.useFakeTimers()
+
+    const { waitFor } = renderHook(() => null)
+
+    let actual = 0
+    const expected = 1
+
+    setTimeout(() => {
+      actual = expected
+    }, 200)
+    let complete = false
+    jest.advanceTimersByTime(200)
+    await waitFor(
+      () => {
+        expect(actual).toBe(expected)
+        complete = true
+      },
+      { timeout: 250, interval: 10 }
+    )
+
+    expect(complete).toBe(true)
+    jest.useRealTimers()
+  })
+  test('should wait for arbitrary expectation to pass when use jest.usedFakeTimers() and runOnlyPendingTimers() ', async () => {
+    jest.useFakeTimers()
+
+    const { waitFor } = renderHook(() => null)
+
+    let actual = 0
+    const expected = 1
+
+    setTimeout(() => {
+      actual = expected
+    }, 200)
+    let complete = false
+    jest.runOnlyPendingTimers()
+    await waitFor(
+      () => {
+        expect(actual).toBe(expected)
+        complete = true
+      },
+      { timeout: 250, interval: 10 }
+    )
+
+    expect(complete).toBe(true)
+    jest.useRealTimers()
```

Comment:
This should be in a `catch` block. If the test fails, we don't want to the fake timers to break other tests in the file.

Reply:
Copy that makes sense. I will wrap the `jest.useRealTimers()` in finally

Reply:
lol, yes, I meant `finally`, not `catch`... putting it in `catch` would not be useful at all 🤦